### ISSUE TITLE
`like` doesn't work with Maybe fields

### ISSUE THREAD

**Issue by rehno-lindeque** [prowdsponsor/esqueleto#68]
It looks like `like` doesn't work with a `Maybe` field in the database because it expects the arguments to satisfy `IsString s`.

E.g. This does not work if `field` of table `A` is a `Maybe` type

```
  results <- select $ from $ \a -> do
      where_ $ (a ^. AField) `like`  concat_ [(%), val "some text", (%)]
      return a
```

**Comment by Pamelloes** [comment]
Just ran into this same issue. I worked out a solution using coalesceDefault, but that seems kind of ridiculous. Has there been any progress lately?

**Comment by AaronFriel** [comment]
I just want to chime in that someone reported having some pain with this in #haskell.

It looks like this is an issue with almost all of the comparison ops, e.g.:

```
(>.) :: PersistField typ => expr (Value typ) -> expr (Value typ) -> expr (Value Bool) 
```

ANSI SQL implementations have well-defined behavior for `nullableColumn > value` and "value > NULL`. I am not sure how it would be typed, but they are valid operations.

**Comment by meteficha** [comment]
@AaronFriel, just use `just`, your case shouldn't have any problem.

**Comment by AaronFriel** [comment]
@meteficha I was trying to help @Pamelloes with this, and they were having a lot of difficulty using `just` correctly. 

Pinging @Pamelloes to share the code snippet, as it's not mine to share and I'm not an esqueleto user myself.

**Comment by Pamelloes** [comment]
The code snippet @AaronFriel provided works if you use just. However when using the like operator, there's an IsString requirement and since there is no IsString instance for IsString a => Maybe a, using just will not work. @mitchellwrosen posted a temporary solution around a year ago, however a more type safe/robust solution should probably exist.

**Comment by meteficha** [comment]
@Pamelloes: `like` is problematic due to `IsString`.  However, `(>.)` should work just fine, which is what @AaronFriel mentioned.

**Comment by Pamelloes** [comment]
@meteficha Exactly :) That's what I was trying to say in my comment though it might not have been as clear as it could be.

**Comment by AaronFriel** [comment]
@meteficha My bad. I wasn't aware the other instances would work. `like` and any other string ops should definitely support operations when the column is nullable.

**Comment by meteficha** [comment]
I've released `esqueleto-2.4.0` with a new `SqlString` class of which `Maybe` is an instance.  Also, there's now `castString`.  Please re-open this issue if you feel something is missing :).
